articles/aks/deploy-spinkube.md

```diff
@@ -0,0 +1,253 @@
+ms.topic: article
+ms.date: 05/27/2024
+author: ThorstenHans
+ms.author: ThorstenHans
+---
+
+# Deploy open-source SpinKube to Azure Kubernetes Service (AKS) to run serverless WebAssembly (Wasm) workloads
+
+[WebAssembly (Wasm)][wasm] is a binary format that is optimized for fast download and maximum execution speed using a Wasm runtime. A Wasm runtime is designed to run on a target architecture and execute WebAssembly modules in a sandbox, isolated from the host computer, at near-native performance. By default, WebAssembly modules can't access resources on the host outside of the sandbox unless it is explicitly allowed, and they can't communicate over sockets to access things like environment variables or HTTP traffic. The [WebAssembly System Interface (WASI)][wasi] standard defines an API for Wasm runtimes to provide access to WebAssembly modules to the environment and resources outside the host using a capabilities model.
+
+[SpinKube][spinkube] is a open-source stack allowing you to run serverless Wasm workloads (Spin Apps) built with open-source [Spin][spin] on top of Kubernetes. In contrast to earlier Wasm runtimes for Kubernetes, Spin Apps are executed natively on the underlying Kubernetes nodes and do not rely on containers. SpinKube consists of four major components: 
+
+- `containerd-shim-spin`: An implementation of [`containerd-shim`][containerd-shim] for Spin, which enables running Spin Apps on Kubernetes via [`run-wasi`][run-wasi]
+- `runtime-class-manager`: A lifecycle operator for managing and automating the lifecycle of `containerd-shims` in a Kubernetes environment (previously known as KWasm)
+- `spin-operator`: A Kubernetes operator allowing the deployment and management of Spin Apps by using custom resources
+- `kube` plugin for `spin`: A `spin` CLI plugin allowing users to scaffold Kubernetes deployment manifests for Spin Apps.
+
+
+## Before you begin
+
+You must have the latest version of Azure CLI, [`kubectl`][kubectl], [Helm][helm], and [Spin CLI][spin-cli] installed.
+
+## Limitations
+
+* The Kubernetes node *os-type* must be Linux.
+* You can't use the Azure portal to deploy SpinKube to an AKS cluster.
+
+## Deploy SpinKube to an existing AKS Cluster
+
+Configure `kubectl` to connect to your Kubernetes cluster using the [az aks get-credentials][az-aks-get-credentials] command. The following command:
+
+```azurecli-interactive
+az aks get-credentials -n myakscluster -g myresourcegroup
+```
+
+### Deploy `cert-manager`
+
+To deploy SpinKube on an AKS cluster, you must have its dependencies installed. SpinKube depends on [cert-manager][cert-manager].
+
+If you haven't deployed `cert-manager` to your AKS cluster yet, you can install it by deploying its Custom Resource Definitions (CRDs), followed by the `cert-manager` Helm chart provided through the `jetstack` repository:
+
+```bash
```

Comment:
Hmm... Might be nice to have these all be `azurecli-interactive` script. I believe it could be done if some env vars were first established. A good example of this is: https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/aks/learn/quick-kubernetes-deploy-cli.md?plain=1.